Small nits to grad accum docs (#656)
huggingface/accelerate: docs/source/concept_guides/gradient_synchronization.mdx

        outputs = model(inputs)
        loss = loss_function(outputs, targets)
        accelerator.backward(loss)
```

> As a result, you should either use *one or the other* when it comes to API choice. 

Reviewer: ```suggestion
As a result, you should either use *`accelerator.accumulate` or `accelerator.no_sync`* when it comes to API choice. 
```
